plugin/evm/vm.go

```diff
@@ -745,6 +723,52 @@ func (vm *VM) writeBackMetadata() {
 	atomic.StoreUint32(&vm.writingMetadata, 0)
 }
 
+// awaitTxPoolStabilized waits for a txPoolHead channel event
+// and notifies the VM when the tx pool has stabilized to the
+// expected block hash
+// Waits for signal to shutdown from txPoolStabilizedShutdownChan chan
+func (vm *VM) awaitTxPoolStabilized() {
+	vm.shutdownWg.Add(1)
```

Comment:
This should be called in the main goroutine, not the spawned goroutine.

Reply:
Updated